### PR TITLE
Add GHC 8.0.1 to travis matrix

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

**Comment by phadej** [comment]
I'm not sure what happens there, filled a bug https://ghc.haskell.org/trac/ghc/ticket/11427#ticket

File: src/Generics/SOP/Constraint.hs [hunk 1]
```diff
@@ -135,5 +138,5 @@ instance
 #if __GLASGOW_HASKELL__ >= 710
   {-# OVERLAPPING #-}
 #endif
-  All SListI xss => SingI (xss :: [[k]]) where
+  (All SListI xss, SListI xss) => SingI (xss :: [[k]]) where
```
Comment (phadej) [inline src/Generics/SOP/Constraint.hs]: See https://ghc.haskell.org/trac/ghc/ticket/11427#comment:4

File: src/Generics/SOP/TH.hs [hunk 2]
```diff
@@ -130,7 +130,11 @@ deriveMetadataValue n codeName datatypeInfoName = do
              , funD datatypeInfoName' [clause [] (normalB $ metadata' isNewtype name cons) []] -- treeDatatypeInfo = ...
              ]
 
+#if MIN_VERSION_template_haskell(2,11,0)
+deriveGenericForDataDec :: Bool -> Cxt -> Name -> [TyVarBndr] -> [Con] -> Cxt -> Q [Dec]
```
Comment (phadej) [inline src/Generics/SOP/TH.hs]: the `Cxt` / `[Name]` parameter isn't used anywhere; I'm not sure what would be wanted approach. This works for now.

**Comment by kosmikus** [comment]
Thanks. Note that I already have a GHC-8 branch. I'll have to compare your changes with mine.

**Comment by kosmikus** [comment]
Ok, that branch is still called 7.11, but still ...

**Comment by phadej** [comment]
@kosmikus Ah, the branches are a bit hidden, hard to spot :) Seems that changes are quite the same; which is good, I guess.

**Comment by kosmikus** [comment]
Yes, CPP magic missing is the main reason I hadn't merged yet.

**Comment by kosmikus** [comment]
I've looked at your version. Except for the export of `-.->`, it looks good to merge. Any idea what we should do about that one? At the very least, it seems like hiding the export should also be conditional, right?

**Comment by phadej** [comment]
@kosmikus I'd wait a bit. Hopefully the regression will be fixed

**Comment by phadej** [comment]
@kosmikus this is now ok to merge. the `-.->` issue is worked around

**Comment by kosmikus** [comment]
Do we need the CPP magic for the `type` keyword? Isn't that backwards compatible with GHC-7?

**Comment by kosmikus** [comment]
I'll merge anyway for now. We can always simplify later.

**Pull request merged**